docs/program/thermal_conductivity.md

```diff
@@ -67,765 +67,302 @@ Optional switches:
-\end{equation}
-$$
+This off-diagonal contribution, describing wavelike-interference between phonons of similar frequencies, becomes important for system with complex unitcell.
+While the derivation sketched here is based on the Hardy current[^Isaeva2019], it can also be obtain from a Wigner description of heat transport [^Simoncelli2019], with very similar results[^Caldarelli2022].
 
+### Monte-Carlo integration for the scattering rates
 
-### Cumulative kappa
+To reach the thermodynamic limit, the thermal conductivity has to be computed on a large grid of q-points, which can make the computation quite expensive.
+This cost comes almost entirely from the computation of the scattering.
 
-@todo Check code snippets
+However, one can observe that the computation of $\kappa$ actually requires two kind of integrations.
+The first is the sum of the contribution of each q-point to the thermal conductivity, while the second one correspond to the computation of the scattering.
 
-@todo Spectral kappa, links to things.
+Fortunately for us, these two integrations converges at different rates.
+In particular, the expensive scattering integration converges more quickly than the thermal conductivity integration.
 
-Experimentally, the cumulative thermal conductivity with respect to phonon mean free path,
+Thus, to improve the computational cost, the code offers the possibility to decouple these two integrations by using a Monte-Carlo integration of the scattering.
+For this, we generate a full grid, on which the thermal conductivity will be integrated.
+A subset of this full grid can then be selected to perform the scattering integration.
+In order to improve the convergence, these point are not selected entirely at random but using a stratified approached in order to sample more uniformly the Brillouin zone.
 
-$$
-l_{\lambda} = \left| v_{\lambda} \right| \tau_{\lambda} \,,
-$$
+This is schematically represented in the following picture, where each dot represents a point on a $8\times8$ grid, with the red dot corresponding to point selected for a Monte-Carlo integration equivalent to a $4\times4$ grid and the bar representing the way the grid is stratified.
 
-can be measured.[^Minnich2012] The cumulative thermal conductivity can then be computed as a sum of the fraction of heat that is carried by phonons with mean free paths smaller than $l$:
+<center>
+<img src="../../media/monte_carlo_grid.png" width="300" />
+</center>
 
-$$
-\kappa_{\alpha\beta}^{\textrm{acc}}(l)=
-\frac{1}{V} \sum_{\lambda}
-C_{\lambda} v^{\alpha}_{\lambda} v^{\beta}_{\lambda} \tau_{\lambda} \Theta(l- l_{\lambda} ) \,,
-$$
+The code allows to use different Monte-Carlo grids for third and fourth order, using the variables `--qpoint_grid3ph` and `--qpoint_grid4ph`.
 
-where $\Theta$ is the Heaviside step function.
+It is important to note that since the points are selected randomly, the results will be noisy.
```

Comment:
I can change this sentence to make it more explicit.
`It is important to note that this method of integration is not deterministic, so that several runs will give different results.`